### ISSUE TITLE
Create toggle component for UserTableFilterDialog

### ISSUE THREAD

**Issue by patcon** [GCTC-NTGC/gc-digital-talent#3381]
Spin-off task of https://github.com/GCTC-NTGC/gc-digital-talent/issues/2641

May be blocked on https://github.com/GCTC-NTGC/gc-digital-talent/issues/3085 (if form state changes, and that PR is processing form state)

<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Description

This was pushed into new ticket by patcon, as was not essential to functionality. Currently, it is being done with a react-select SelectField with one value.

## Screenshot: Prototype
[![a screenshot of ...][image]][image-link]
<sup>[Click screenshot to visit Figma.][image-link]</sup>

## Acceptance Criteria
- [ ] ...

   [image]: https://user-images.githubusercontent.com/305339/179536912-1c718120-7257-428e-9e2c-cf072e17b878.png
   [image-link]: https://example.com
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

**Comment by patcon** [comment]
I'm wondering if we could make the form field more similar to other fields, ie. with the label smaller and above instead of beside. Not sure if it's helpful to minimize bespoke markup (like if a toggle needs context, we don't need to rethink how that's displayed differently? Also would seem to allow smaller width, as they can wrap onto two lines and label is smaller.)

**Comment by gobyrne** [comment]
@patcon we have this component now, right?

**Comment by patcon** [comment]
I know there's ToggleGroup with single-select, but I recall some mention of it not being a suitable use for a checkbox-like toggle? cc @esizer 
https://main--61e099a1bb1465003a73d3ce.chromatic.com/?path=/story/components-togglegroup--single

**Comment by esizer** [comment]
> I know there's ToggleGroup with single-select, but I recall some mention of it not being a suitable use for a checkbox-like toggle?

@patcon you are correct, I was going to make it using [@radix-ui/react-switch](https://www.radix-ui.com/docs/primitives/components/switch). I can squeeze it into #4136 unless it is higher priority then I can make it today in a separate PR 😄

**Comment by patcon** [comment]
Thanks @esizer! I've re-ticketed this to a clearer issue for actual implementation, and close this as done :) https://github.com/GCTC-NTGC/gc-digital-talent/issues/5115
